### PR TITLE
FIM Caching Fixes

### PR DESCRIPTION
**Pull request by nickchadwick-noaa**
There were a small handful of Terraform Syntax and Logic errors that I needed to fix before deploying all the new Redshift resources.

Additionally, I included a bugfix that hopefully makes the image-based Lambdas update correctly when their underlying code changes.

### CODE REVIEW

File: Core/IAM/Roles/redshift.json.tftpl
Comment (nickchadwick-noaa) [inline Core/IAM/Roles/redshift.json.tftpl:1]: No underscored allowed in IAM Policies.

File: Core/EC2/RDSBastion/scripts/viz/redshift_setup.sh.tftpl
Comment (nickchadwick-noaa) [inline Core/EC2/RDSBastion/scripts/viz/redshift_setup.sh.tftpl:1]: I think I set these usernames and passwords correctly? You may need to doublecheck.

File: Core/LAMBDA/viz_functions/viz_publish_service/lambda_function.py
Comment (nickchadwick-noaa) [inline Core/LAMBDA/viz_functions/viz_publish_service/lambda_function.py:1]: Small fix for an edge-case where EGIS messes up and provides garbage data win the list of services.

File: Core/EC2/RDSBastion/scripts/viz/redshift_setup.sh.tftpl
Comment (TylerSchrag-NOAA) [reply Core/EC2/RDSBastion/scripts/viz/redshift_setup.sh.tftpl:1]: Thank you! I totally missed this.

**Review by TylerSchrag-NOAA — APPROVED** [review]
Thanks so much for catching / fixing these issues, Nick!

**Pull request merged**